### ISSUE TITLE
Model Simulation Based Inference

### ISSUE THREAD

**Issue by hsimonfroy** [hsimonfroy/montecosmo#4]
# Context
Documenting field-level Simulation Based Inference from a differentiable cosmological model.

In [code], we run joint inferences of the initial field ($\delta_L$, $64^3$ mesh), cosmological parameters ($\Omega_c$ and $\sigma_8$), and Lagrangian bias parameters.  

# Aims

- [ ] Dataset generation
- [ ] Compression
- [ ] NDE

**Comment by Justinezgh** [comment]
Here are some useful links: 

- **Create a TensorFlow dataset** (optional; example [here](https://github.com/DifferentiableUniverseInitiative/sbi_lens/blob/main/sbi_lens/gen_dataset/lensing_lognormal_dataset.py))

- **Compression** (example [here](https://github.com/DifferentiableUniverseInitiative/sbi_lens/blob/main/scripts/train_compressor.py))
 This part aims to compress your data into summary statistics with dim(summary statistics) = dim(parameters you want to infer).

     - chose an architecture 
     → I usually use [ResNet 18](https://github.com/google-deepmind/dm-haiku/blob/main/haiku/_src/nets/resnet.py)


     - chose a loss function 
     → we can start with vmim ([the paper](https://arxiv.org/abs/2009.08459))

     - check what data augmentation we can do
      → I usually add the noise on the fly and do rotation & flipping (example [here](https://github.com/DifferentiableUniverseInitiative/sbi_lens/blob/main/sbi_lens/gen_dataset/utils.py))

- **Inference**
Based on the previous compression you now what to infer your parameters.
→ NPE should be good  (example [here](https://gist.github.com/Justinezgh/75ee6fbd05c999f81b2d18d61f4fc308))
